project/api/models.py

```diff
@@ -0,0 +1,79 @@
+"""
+Chron.
+Copyright (C) 2018 Alisa Belyaeva, Ata Ali Kilicli, Amaury Martiny,
+Daniil Mordasov, Liam O’Flynn, Mikhail Orlov.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+from django.db import models
+from polymorphic.models import PolymorphicModel
+
+from .ltree import LtreeField
+
+# Create your models here.
+class PoliticalEntity(models.Model):
+    """
+    A 1-1 mapping between a https://www.wikidata.org/wiki/Q56061, and a PK in our db.
+    Holds an additional color information.
+    """
+
+    wikidata_id = models.IntegerField(unique=True)  # Excluding the Q
+    color = models.IntegerField()
```

Comment:
color should be an rgb field. This is the "wish" color that's inputted by mappers.

Reply:
Ah, I was thinking a post-save hook that will run the algo and generate it in the generated tables. @MiklerGM thoughts?

Reply:
imo storing final color is not needed in PE table if we already have the `predecessor` field.

That's a great link, I agree to put an array of ColorField too. Too bad this info is not in wikidata.

Reply:
Can you describe how do you see workflow of a coloring algorithm? Each run is a dry-run or it should reuse some previous gathered data or this data would be stored separately somewhere? If so, there do not need to store final color in PE table